src/yang/ietf-scheduler-policy.yang

```diff
@@ -59,49 +59,57 @@ module ietf-scheduler-policy {
     when "derived-from-or-self(../../../policy:type, " +
          "'qos-types:scheduler-policy-type')" {
       description
-        "Only when policy type is scheduler-policy";
-      }
+        "This filter can be used only when policy type is
+         scheduler-policy.";
+    }
     choice filter-params {
       description
-        "Choice of action types";
+        "Choice of filter parameters.";
       case filter-match-all {
         uses diffserv:filter-match-all;
         description
-          "filter match-all";
+          "A filter that matches all packets.";
       }
     }
     description
-      "Augments Queue policy Classifier with common filter types";
+      "Augments Queue Policy Classifier to add filter parameters.";
   }
 
   augment "/policy:policies/policy:policy/policy:classifier" + 
           "/policy:action/policy:action-params" {
     when "derived-from-or-self(../../policy:type, " +
          "'qos-types:scheduler-policy-type')" {
       description
-        "Augments only when policy type is scheduler-policy.";
+        "These actions apply only when policy type is
+         scheduler-policy.";
     }
-    case scheduler {
-      uses action:scheduler;
-    }
-    case queue-policy-name {
-      container queue-policy-name {
-        leaf queue-policy {
-          type leafref  {
-            path "/policy:policies/policy:policy/policy:name";
+    choice action {
```

Comment:
action-params in the augment path is a choice statement.. I am wondering why it will be required..

Reply:
Because you had two case statements. If this is not a choice/case situation, we can remove both choice and case statements.

Reply:
I mean, we are already augmenting to choice statement in ietf-traffic-policy (action-params), hence it was only case statements, no choice statements.

Reply:
I think original idea was A,B,C + D,E,F = A,B,C,D,E,F i.e. one XML can include configurations from both modules..

Reply:
I do not think what you are doing is A,B,C + D,E,F = A,B,C,D,E,F.

You have ietf-diffserv which is augmenting and adding A,B,C
Then you have ietf-queue-policy which is augmenting the same node and adding D,E,F
and then you have the scheduler-policy which is augmenting G,H,I

but they are all different modules, and it cannot be under the the same tree hierarchy.

Reply:
Yes, that is what we understood, it doesn't add up that way.. infact, in same module we couldn't do two separate augmentation to same tree point and hence diffserv, queue and scheduler have to be different modules..